### ISSUE TITLE
Tree filter with more than 10 nodes

### ISSUE THREAD

**Issue by melloware** [primefaces/primefaces#2301]
From this forum thread:  https://forum.primefaces.org/viewtopic.php?f=3&t=50532

## 1) Environment
- PrimeFaces version: 6.1
- Does it work on the newest released PrimeFaces version? Version? NO
- Does it work on the newest sources in GitHub? NO
- Application server + version: JBOSS EAP 6.4
- Affected browsers: ALL

## 2) Expected behavior

Tree Filtering works for more than 10 nodes

## 3) Actual behavior

The check:

`if(rk.startsWith(rowKey) || rowKey.startsWith(rk))`

is wrong in my case there is a node that was matched with id: 0_101 and this check match also node with other ids e.g.: 0_1

I understand the logic of this check (matching parents of matched nodes) but it doesn't work in this case and in general I don't thing its a very good idea to put to many assumptions and logic into the id structure. The id should be unique and 'dumb' so it could be overridden easily by applications.

## 4) Steps to reproduce

Create a tree with many nodes and try to filter.

Suggested fix from @tefron:
```java
        if (filter && filteredRowKeys.size() > 0) {
            for (String rk : filteredRowKeys) {
                String rowKeyExt = rowKey + "_";
                String rkExt = rk + "_";
                if (rk.startsWith(rowKeyExt) || rowKey.startsWith(rkExt) || rk.equals(rowKey)) {
                    match = true;
                    if (!node.isLeaf() && !rowKey.startsWith(rk)) {
                        node.setExpanded(true);
                    }
                    break;
                }
            }
```

A PR will follow.

**Comment by LeandriT** [comment]
can someone tell us, if this throuble is already solved?? I tried with newer versión of primefaces and i can´t filter on p tree component...

**Comment by cnsgithub** [comment]
~~@LeandriT As you can see above, the issue has been resolved by @melloware in https://github.com/primefaces/primefaces/pull/2302 and merged by @tandraschko. However, there is not yet a released version that contains the fix. All you can do is to build PrimeFaces by your own or wait for a new version.~~

Didn't see the milestone label, sorry. I have to go to the ophthalmologist again. :eye:

**Comment by melloware** [comment]
@LeandriT This was fixed for PF 6.2 can you not upgrade?
